solidity/contracts/CosmosToken.sol

```diff
@@ -1,16 +1,78 @@
-pragma solidity ^0.6.6;
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+pragma solidity ^0.8.0;
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Capped.sol";
 
-contract CosmosERC20 is ERC20 {
+contract CosmosERC20 is ERC20Burnable, ERC20Capped {
 	uint256 MAX_UINT = 2**256 - 1;
 
+	address gravity;
+
 	constructor(
 		address _gravityAddress,
 		string memory _name,
 		string memory _symbol,
-		uint8 _decimals
-	) public ERC20(_name, _symbol) {
+		uint8 _decimals,
+		uint256 _maxSupply
+	) public ERC20(_name, _symbol) ERC20Capped(_maxSupply) {
```

Comment:
I need to understand what ERC20Capped does better. 

There won't be a way to know the max possible supply of the Cosmos side coin. My mental model is the initial supply of all of these coins should be zero when the factory is run and then goes up as batches of outgoing txs create coins.

Reply:
That was based on my understanding that we have a 500mm max supply for SOMM - so it's a safety mechanism to make sure the minted supply is never above that. It can be trivially removed if that max supply shouldn't be enforced.

Reply:
This will be effectively any arbitrary token. The cosmos state machines should have control to mint tokens as it desires.